src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -174,15 +174,20 @@ private void loadFile(File fileToLoad) {
   public void buttonLoadConfigClicked() {
     if (checkForSaveAndContinue()) {
       FileChooser fc = new FileChooser();
+
+      GuiContext.getInstance().fileChooser = fc;
+
       if (selectedFile == null) {
         fc.setInitialDirectory(new File(System.getProperty("user.dir")));
       } else {
         fc.setInitialDirectory(new File(selectedFile.getParent()));
       }
       fc.getExtensionFilters().add(new ExtensionFilter("JSON files", "*.json"));
       fc.setTitle("Load Config");
+      File fileToLoad = fc.showOpenDialog(GuiContext.getInstance().mainWindow);
```

Comment:
Instead of needing to do all that jazz in GuiContext to put the window here, I'm pretty sure we can reference the window that contains the button being clicked. Maybe it's cleaner to do it via GuiContext though? Not sure, would need to play around with the other option and see how it looks.

Reply:
dig!

Reply:
unfortunately there is no easier path from within the event handler to obtain the root window so I went ahead with plan A.